tensorflow/python/ops/array_ops.py

```diff
@@ -99,6 +99,21 @@
 listdiff = gen_array_ops.list_diff
 
 
+def rank(input, name=None):
+  """Override for gen_array_ops.rank(), making it work for SparseTensor.
+
+  Args:
+    input: A `Tensor` or `SparseTensor` object.
+    name: A name for the operation (optional).
+
+  Returns:
+    A `Tensor` of type `int32`.
+  """
+  if isinstance(input, ops.SparseTensor):
```

Comment:
It's customary to wrap everything inside a [`ops.op_scope()`](https://www.tensorflow.org/versions/r0.8/api_docs/python/framework.html#op_scope) in a Python op.